macaroonbakery/tests/test_client.py

```diff
@@ -55,6 +55,35 @@ def handler(*args):
         finally:
             httpd.shutdown()
 
+    def test_cookie_domain_host_not_fqdn(self):
+        b = new_bakery('loc', None, None)
+
+        def handler(*args):
+            GetHandler(b, None, None, None, None, AGES, *args)
+        try:
+            httpd = HTTPServer(('', 0), handler)
+            thread = threading.Thread(target=httpd.serve_forever)
+            thread.start()
+            srv_macaroon = b.oven.macaroon(
+                version=bakery.LATEST_VERSION, expiry=AGES,
+                caveats=None, ops=[TEST_OP])
+            self.assertEquals(srv_macaroon.macaroon.location, 'loc')
+            client = httpbakery.Client()
+            resp = requests.get(
```

Comment:
# Note: by using "localhost" instead of the presumably numeric address held
    # in httpd.server_address, we're triggering the no-FQDN logic in the cookie
    # code.

?